var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -52,13 +53,13 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
 
     # ROCm 4.1.0 breaks platform settings which breaks Ginkgo's HIP support.
     conflicts("^hip@4.1.0:", when="@:1.3.0")
-    conflicts("^hip@4.1.0:", when="@master")
     conflicts("^hipblas@4.1.0:", when="@:1.3.0")
-    conflicts("^hipblas@4.1.0:", when="@master")
     conflicts("^hipsparse@4.1.0:", when="@:1.3.0")
-    conflicts("^hipsparse@4.1.0:", when="@master")
     conflicts("^rocthrust@4.1.0:", when="@:1.3.0")
-    conflicts("^rocthrust@4.1.0:", when="@master")
+
+    # Skip smoke tests if compatible hardware isn't found
+    patch('1.4.0_skip_invalid_smoke_tests.patch', when='@master')
+    patch('1.4.0_skip_invalid_smoke_tests.patch', when='@1.4.0')
```

Comment:
Did you create this patch explicitly for Spack, or can it be downloaded form some url?

Reply:
This is explicitly for spack. In later releases, this should not be needed though.